legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/model/DataSpaceTemplateExecutablePointerInfo.java

```diff
@@ -0,0 +1,22 @@
+// Copyright 2024 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.generation.analytics.model;
+
+public class DataSpaceTemplateExecutablePointerInfo extends DataSpaceExecutableInfo
+{
+    public String id;
+    public String executionContextKey;
+    public String function;
```

Comment:
could we rename this to `executable` or packagablePointer. 
We could add support for other elements (service etc) in the future.

Reply:
sounds good will do. But won't it overlap with the existing dataspace executables that already support service?